statsmodels/multivariate/multivariate_ols.py

```diff
@@ -0,0 +1,517 @@
+    `name` is the effect (i.e. term in model) name and `df` is a DataFrame
+    containing the test statistics
+
+    """
+    def __init__(self, fitted_mv_ols):
+        if (hasattr(fitted_mv_ols, 'data') and
+                hasattr(fitted_mv_ols.data, 'design_info')):
+            self.design_info = fitted_mv_ols.data.design_info
+        else:
+            self.design_info = None
+        self.exog_names = fitted_mv_ols.exog_names
+        self.endog_names = fitted_mv_ols.endog_names
+        self._fittedmod = fitted_mv_ols._fittedmod
+
+    def __str__(self):
+        return self.summary().__str__()
+
+    def __getitem__(self, item):
+        return self.results[item]
+
+    def mv_test(self, hypotheses=None):
+
+        k_xvar = len(self.exog_names)
+        if hypotheses is None:
+            if self.design_info is not None:
+                terms = self.design_info.term_name_slices
+                hypotheses = []
+                for key in terms:
+                    L_contrast = np.eye(k_xvar)[terms[key], :]
+                    hypotheses.append([key, L_contrast, None])
+            else:
+                hypotheses = []
+                for i in range(k_xvar):
+                    name = 'x%d' % (i)
+                    L = np.zeros([1, k_xvar])
+                    L[i] = 1
+                    hypotheses.append([name, L, None])
+
+        results = _multivariate_ols_test(hypotheses, self._fittedmod,
+                                          self.exog_names, self.endog_names)
+
+        return MultivariateTestResults(results)
+    mv_test.__doc__=(
+        """
+        Testing the linear hypotheses
+            L * params * M = C
+        where `params` is the regression coefficient matrix for the
+        linear model y = x * params, `L` is the contrast matrix, `M` is the
+         dependent variable transform matrix and C is the constant matrix.
+
+        Parameters
+        ----------
+        """ + _hypotheses_doc +
+        """
+        Returns
+        -------
+        results: _MultivariateOLSResults
+
+        """)
+    def summary(self):
+        raise NotImplementedError
+
+
+class MultivariateTestResults(object):
+    def __init__(self, mv_test_df):
+        self.results = mv_test_df
+
+    def __str__(self):
+        return self.summary().__str__()
+
+    def __getitem__(self, item):
+        return self.results[item]
+
+    @property
+    def summary_frame(self):
+        """
+        Return results as a multiindex dataframe
+        """
+        df = []
+        for key in self.results:
+            tmp = self.results[key]['stat'].copy()
+            tmp.loc[:, 'Effect'] = key
+            df.append(tmp.reset_index())
+        df = pd.concat(df, axis=0)
+        df = df.set_index(['Effect', 'index'])
+        df.index.set_names(['Effect', 'Statistic'], inplace=True)
+        return df
+
+    def summary(self, contrast_L=False, transform_M=False):
```

Comment:
the names of the keywords are confusing, maybe prefix with "show"
`show_contrast_L=False, show_transform_M=False`

My first impression when reading this was that I need to provide the L and M matrices.